src/component/mxgraph/shape/EndEventShape.ts

```diff
@@ -17,16 +17,58 @@
 import { MxGraphFactoryService } from '../../../service/MxGraphFactoryService';
 import { mxgraph } from 'ts-mxgraph';
 import { StyleConstant } from '../StyleConfigurator';
+import { ShapeBpmnEventKind } from '../../../model/bpmn/shape/ShapeBpmnEventKind';
 
 const mxEllipse: typeof mxgraph.mxEllipse = MxGraphFactoryService.getMxGraphProperty('mxEllipse');
+const mxUtils: typeof mxgraph.mxUtils = MxGraphFactoryService.getMxGraphProperty('mxUtils');
+const mxConstants: typeof mxgraph.mxConstants = MxGraphFactoryService.getMxGraphProperty('mxConstants');
 
-export default class EndEventShape extends mxEllipse {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public constructor(bounds: mxgraph.mxRectangle, fill: any, stroke: any, strokewidth = StyleConstant.STROKE_WIDTH_THICK) {
+class EventShape extends mxEllipse {
+  public constructor(bounds: mxgraph.mxRectangle, fill: string, stroke: string, strokewidth: number) {
     super(bounds, fill, stroke, strokewidth);
   }
 
   public paintVertexShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
+    this.paintOuterShape(c, x, y, w, h);
+    this.paintInnerShape(c, x, y, w, h);
+  }
+
+  protected paintOuterShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
     super.paintVertexShape(c, x, y, w, h);
   }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected paintInnerShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
+    // do nothing by default
+  }
+}
+
+export default class EndEventShape extends EventShape {
+  public constructor(bounds: mxgraph.mxRectangle, fill: string, stroke: string, strokewidth: number = StyleConstant.STROKE_WIDTH_THICK) {
+    super(bounds, fill, stroke, strokewidth);
+  }
+
+  protected paintInnerShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
+    const eventKind = mxUtils.getValue(this.style, StyleConstant.BPMN_STYLE_EVENT_KIND, ShapeBpmnEventKind.NONE);
+    if (eventKind == ShapeBpmnEventKind.TERMINATE) {
+      this.paintTerminateEventIcon(c, x, y, w, h);
+    }
+  }
+
+  // highly inspired from mxDoubleEllipse
+  private paintTerminateEventIcon(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
```

Comment:
**_A note about wording used for this method_**
The BPMN spec both uses the `marker` (most of the time) and `icon` wording when it speaks about the graphical element that identifies the event type (terminate, cancel, ....).
_Bruce Silver's "Bpmn & Style"_ book uses `icon` and uses `marker` when describing task supplemental behaviour (loop, compensation, ...).

I chose to use `icon` as _Bruce Silver_ do,  to avoid ambiguity and to have the same naming for all shapes in the future.
Feel free to change, if you disagree